core/src/test/java/org/elasticsearch/indices/store/IndicesStoreIntegrationTests.java

```diff
@@ -69,6 +73,7 @@ protected Settings nodeSettings(int nodeOrdinal) { // simplify this and only use
                 // which is between 1 and 2 sec can cause each of the shard deletion requests to timeout.
                 // to prevent this we are setting the timeout here to something highish ie. the default in practice
                 .put(IndicesStore.INDICES_STORE_DELETE_SHARD_TIMEOUT, new TimeValue(30, TimeUnit.SECONDS))
+                .put(TransportModule.TRANSPORT_SERVICE_TYPE_KEY, MockTransportService.class.getName())
```

Comment:
cluster state updates are now only blocked between relocation start and end and relocation does not seem to need a new cluster state to finish. if another cluster state is published it will only be delayed for as long as relocation takes. the test now takes about 2s.